connector/connect/common/src/main/protobuf/spark/connect/base.proto

```diff
@@ -183,6 +183,85 @@ message ExecutePlanResponse {
   }
 }
 
+// Request to update or fetch the configurations.
+message ConfigRequest {
```

Comment:
maybe more like

```
message ConfigRequest {
   oneof type {
      GetConfig get_config = 1;
      SetConfig set_config = 2;
   }

   message GetConfig {
      repeated string keys = 1;
   }

   message SetConfig {
      repeated KeyValue pairs = 1;
   }
}



```

Reply:
I think it might be good to have explicit messages rather than the combination of the enum plus the message attributes.